### ISSUE TITLE
Coaching a friend after spectating the game live throws internal error

### ISSUE THREAD

**Issue by tinvernizzi** [jeffhill/Dota2#187]
### Description

# Description

Coaching does not work if you spectated the game prior to trying to coach.

# Steps to reproduce

1 - Right click someone playing in your friend list
2 - Click "Watch Friend Live"
3 - After loading in game, go to the main menu and click disconnect
4 - Right click the same friend you just spectated
5 - Click "Coach Friend"
6 - You should see the error message in the screenshot

### Screenshots

![image](https://user-images.githubusercontent.com/13117941/171724920-01562df2-cbe2-4fcf-8f63-748621672e12.png)

**Comment by NMOTH** [comment]
Are you trying to coach in ranked game?  you cant coach in ranked game.

**Comment by tinvernizzi** [comment]
> Are you trying to coach in ranked game? you cant coach in ranked game.

That's not true. You can coach a friend in a ranked game by right clicking on them and selecting the "Coach Friend" option.
